### PR TITLE
code review

### PR DESCRIPTION
**Pull request by Steven-Sorokin**

### CODE REVIEW

File: ArucoDetectRos-Amir's_Changes.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/env python3
+from __future__ import division  #in order to be able to set the (1/30)
+"""
+Framework   : OpenCV Aruco
+Description : Calibration of camera and using that for finding pose of multiple markers
+Status      : Working
```
Comment (Steven-Sorokin) [inline ArucoDetectRos-Amir's_Changes.py:6]:
```suggestion
Status      : Working
```

File: ArucoDetectRos-Amir's_Changes.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/env python3
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:1]:
```suggestion
#!/usr/bin/env python
```
You should use python2.7 when using rospy.

**Review by trakevital — CHANGES_REQUESTED** [review]

File: ArucoDetectRos-Amir's_Changes.py [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+Description : Calibration of camera and using that for finding pose of multiple markers
+Status      : Working
+References  :
+    1) https://docs.opencv.org/3.4.0/d5/dae/tutorial_aruco_detection.html
+    2) https://docs.opencv.org/3.4.3/dc/dbb/tutorial_py_calibration.html
+    3) https://docs.opencv.org/3.1.0/d5/dae/tutorial_aruco_detection.html
+"""
+
+from math import floor
+import numpy as np
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:14]: What should be the standard when importing numpy?

1. Should we do `import numpy as np`
2. Or should we leave it as `import numpy`?

File: ArucoDetectRos-Amir's_Changes.py [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+from cv_bridge import CvBridge
+from sensor_msgs.msg import Image
+from sensor_msgs.msg import CameraInfo
+from sensor_msgs.msg import Imu
+import tf
+from std_msgs.msg import UInt32
+from std_msgs.msg import String
+import message_filters
+import tf.transformations as tr
+import numpy as np
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:26]:
```suggestion
```
This is a duplicate from line 14

File: ArucoDetectRos-Amir's_Changes.py [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+import pickle
+import glob
+import cv2
+import cv2.aruco as aruco
+import csv
+import tf2_ros
+
+calib_path_param = '/home/makeruser/wifi-Project/Aruco_Tracker/images/for_calib/*.jpg'  ########## AMIR -> change these images and location
+aruco_dict_param = aruco.DICT_4X4_250 ########## AMIR -> this work for https://chev.me/arucogen/
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:35]: There are TODO comments in here which should be turned into issues/tasks or just removed
```suggestion
calib_path_param = '/home/makeruser/wifi-Project/Aruco_Tracker/images/for_calib/*.jpg'  
aruco_dict_param = aruco.DICT_4X4_250 
```

File: ArucoDetectRos-Amir's_Changes.py [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+marker_length = 0.19 # meters
+
+# floor_ids = [101,102]
+# human_ids = [100, 105]
+floor_ids = [102, 103, 104]
+human_ids = [0, 101, 1, 100]
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:41]: Constant variables in python should be all-caps
```suggestion
CALIB_PATH_PARAM = '/home/makeruser/wifi-Project/Aruco_Tracker/images/for_calib/*.jpg'  ########## AMIR -> change these images and location
ARUCO_DICT_PARAM = aruco.DICT_4X4_250 ########## AMIR -> this work for https://chev.me/arucogen/
MARKER_LENGTH = 0.19 # meters

FLOOR_IDS = [102, 103, 104]
HUMAN_IDS = [0, 101, 1, 100]
```

File: ArucoDetectRos-Amir's_Changes.py [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+
+####---------------------- CALIBRATION ---------------------------
+def calib_camera(calib_path=calib_path_param):
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:44]: Function has no doc-string

File: ArucoDetectRos-Amir's_Changes.py [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+calib_path_param = '/home/makeruser/wifi-Project/Aruco_Tracker/images/for_calib/*.jpg'  ########## AMIR -> change these images and location
+aruco_dict_param = aruco.DICT_4X4_250 ########## AMIR -> this work for https://chev.me/arucogen/
+marker_length = 0.19 # meters
+
+# floor_ids = [101,102]
+# human_ids = [100, 105]
+floor_ids = [102, 103, 104]
+human_ids = [0, 101, 1, 100]
+
+####---------------------- CALIBRATION ---------------------------
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:43]: remove usesless line
```suggestion
```

File: ArucoDetectRos-Amir's_Changes.py [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+def calib_camera(calib_path=calib_path_param):
+    # termination criteria for the iterative algorithm
+    criteria = (cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_MAX_ITER, 30, 0.001)
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:46]: `criteria` isn't a significant name for a variable.
```suggestion
    iter_algo_termination_criteria = (cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_MAX_ITER, 30, 0.001)
```

File: ArucoDetectRos-Amir's_Changes.py [hunk 10]
```diff
@@ -0,0 +1,349 @@
...
+
+    # prepare object points, like (0,0,0), (1,0,0), (2,0,0) ....,(6,5,0)
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:48]: What does this comment mean? I don't understand what it is talking about.

File: ArucoDetectRos-Amir's_Changes.py [hunk 11]
```diff
@@ -0,0 +1,349 @@
...
+    # checkerboard of size (7 x 6) is used
+    objp = np.zeros((6*7, 3), np.float32)
+    objp[:,:2] = np.mgrid[0:7, 0:6].T.reshape(-1, 2)
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:51]: All of this needs to be in a constant variable. For example:
```python
C_BOARD_COLS = 6
C_BOARD_ROWS = 7
CHECKERBOARD_INFO = (C_BOARD_COLS*C_BOARD_ROWS, 3)
```

File: ArucoDetectRos-Amir's_Changes.py [hunk 12]
```diff
@@ -0,0 +1,349 @@
...
+
+    # arrays to store object points and image points from all the images.
+    objpoints = [] # 3d point in real world space
+    imgpoints = [] # 2d points in image plane.
```
Comment (trakevital) [inline ArucoDetectRos-Amir's_Changes.py:55]: hard to understand the variable names `objpoints` and `imgpoints`, also not in conventions! Needs `_` as separator between words:
```suggestion
    real_world_points = [] 
    imgage_points = [] 
```

**Review by trakevital — CHANGES_REQUESTED** [review]

**Pull request closed without merging**